src/main/java/com/teenthofabud/wizard/nandifoods/wms/settings/unit/uom/service/impl/UOMServiceImpl.java

```diff
@@ -187,8 +193,10 @@ private void crossLink(UOMEntity from, Optional<List<UnitClassCrossLinkageForm>>
             return;
         }
         List<UnitClassCrossLinkageForm> linkedPUHUs = optionalUnitClassCrossLinkageForms.get();
-        linkedPUHUs.stream().filter(e -> e.getType().compareTo(UnitClass.PU) == 0).forEach(e -> puLink(from, e));
-        linkedPUHUs.stream().filter(e -> e.getType().compareTo(UnitClass.HU) == 0).forEach(e -> huLink(from, e));
+        linkedPUHUs.stream().filter(e -> e.getType().compareTo(UnitClass.PU) == 0).forEach(Errors.rethrow().wrap(e -> {
+            puLink(from, e);
```

Comment:
combine PU and HU comparison together within the same stream using ternary operators